lib/task/import/csvImportTask.class.php

```diff
@@ -956,16 +962,35 @@ public function execute($arguments = array(), $options = array())
         // if both a URI and path are provided, the former is preferred.
         if ($uri = $self->rowStatusVars['digitalObjectURI'])
         {
+          $do = new QubitDigitalObject;
+          $do->informationObject = $self->object;
+
+          if (isset($self->status['options']['skip-derivatives']))
+          {
+            // If not creating derivatives, then skip download altogether
+            $options = array('skipDerivatives' => true);
+          }
+          else
+          {
+            // Try downloading external object up to three times (2 retries)
+            $options = array('downloadRetries' => 2);
+          }
+
+          // Catch digital object import errors to avoid killing whole import
           try
           {
-            if (!downloadExternalURIWithRetries($uri, $self->object))
-            {
-              $this->log("Failed to download $uri after multiple tries. Continuing task...");
-            }
+            $do->importFromURI($uri, $options);
+            $do->save($conn);
           }
-          catch (Exception $e)
+          catch (sfException $e)
           {
-            print "\nFailed to download '$uri': ". $e->getMessage() ."\n";
+            echo $e->getMessage()."\n";
```

Comment:
Agreed. For some reason I thought $this->log was not in scope, but I was wrong.  It's annoying that $this->log() only echoes to STDOUT and doesn't actually write to the *log* file.  I thought for a minute I could ditch the subsequent code for writing the error to the log. :(